### PR TITLE
Validation for IPWithHostnameField

### PR DESCRIPTION
**Pull request by zefciu**
The IPWithHostname field should always require IP Address, or else the
IPAddress object creation will fail.

### CODE REVIEW

File: src/ralph/ui/forms/addresses.py [hunk 1]
```diff
@@ -388,6 +389,9 @@ def __init__(self, *args, **kwargs):
     widget = IPWithHostWidget()
 
     def compress(self, value):
+        hostname, ip_number = value
+        if not ip_number:
+            raise ValidationError(_('IP Address is required'))
```
Comment (xliiv) [inline src/ralph/ui/forms/addresses.py]: I know this works, but more reasonable place for this code is clean(..), I think.

**Comment by mkurek** [comment]
I've checked out your branch and test it and this form no longer throw 500 when saving None IPAddress, BUT - despite the fact that you are throwing ValidationError user don't get any message that something goes wrong and ip field is not mark with ex. red color to suggest that there was a problem with this field. What's worse, it's looks like this form is silently ignoring ValidationError and after submitting it, it present values before changes. 

ip_host_valid

Besides that when i'm trying to save empty form i get an error: ValueError: need more than 0 values to unpack: models_device.py:810

File: src/ralph/ui/forms/addresses.py [hunk 2]
```diff
@@ -388,6 +389,10 @@ def __init__(self, *args, **kwargs):
     widget = IPWithHostWidget()
 
     def compress(self, value):
+        if value:
+            hostname, ip_number = value
+            if not ip_number:
+                raise ValidationError(_('IP Address is required'))
```
Comment (kula1922) [inline src/ralph/ui/forms/addresses.py]: why not
if value and not value[1]:

File: src/ralph/ui/tests/unit/test_fields.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+from ralph.ui.forms.addresses import IPWithHostField
+
+
+class TestIPWithHostnameField(TestCase):
+    """Tests for ``IPWithHostField``"""
+
+    def test_field_invalid_when_ip_number_missing(self):
+        field = IPWithHostField()
+        with self.assertRaises(ValidationError):
+            field.compress(['example.com', ''])
```
Comment (kula1922) [inline src/ralph/ui/tests/unit/test_fields.py]: I am not sure but it should work too

with self.assertRaises(ValidationError):
            IPWithHostField().compress(['example.com', ''])

File: src/ralph/ui/forms/addresses.py (hunk 2 above)
Comment (vi4m) [inline src/ralph/ui/forms/addresses.py]: No docstring is a much important issue!

Comment (kula1922) [inline src/ralph/ui/forms/addresses.py]: There is not need know about veriables naming.  _('IP Address is required') give you everything what you need to know. 

Are you thinking that two nested "if" statement are little low shitdifficult to understand?

Comment (zefciu) [inline src/ralph/ui/forms/addresses.py]: I prefer not to address tuples with index numbers as they are meaningless. I believe this is not overnested here.l

Comment (zefciu) [inline src/ralph/ui/forms/addresses.py]: The ``compress`` is a standard method on this field. It is defined by django and need not to be documented by me.

**Comment by vi4m** [comment]
LGTM

**Comment by mkurek** [comment]
:ok:

**Pull request merged**